### DISCUSSION TITLE
Using ListRandomizer for balanced conditions when sample size is not known

### DISCUSSION THREAD

**Discussion by markusgermar** [ctreffe/alfred#246] [Alfred Q&A] [answered]
Hi,

I would like to use ListRandomizer in order to arrive at conditions with equal sample size. However, I do not know the total sample size of my study (I will collect data from as many participants as possible during one day).

How do I use ListRandomizer (or another function) in order to arrive at the same (or approximately the same) number of participants per conditions in this case? (There will be two conditions.)

Thanks!
Markus

**Comment by jobrachem** [comment, marked as answer]
If you do not know the target sample size, I think you basically have three options:

1. Use the ListRandomizer with batches. That means, in a first version of the experiment, you assign only 10 people in total, 5 of which go to the control group and 5 go to the experimental group. Then you increase the version number to get the next batch and so on. This requires quite a bit of manual intervention though.
2. Use alfred3_interact with a [sequentialspec](https://github.com/jobrachem/alfred3-interact/blob/main/src/alfred3_interact/spec.py#L296). In this case you would "abuse" the sequential pairing of dyads simply to make sure that the experiment always progresses in pairs of (control / experiment). 
3. Use [random_condition](https://alfredo3.psych.bio.uni-goettingen.de/docs/generated/alfred3.randomizer.random_condition.html) for actual pseudo-random assignment - although this is likely to lead to imbalanced results.

If you are fine with the slight conceptual limit to the randomness in option 2, I would recommend that one.

**Reply by jobrachem** [reply]
P.S.: Here's the link to the [SequentialSpec docs](https://jobrachem.github.io/alfred3-interact/build/html/generated/alfred3_interact.spec.SequentialSpec.html)
